### ISSUE TITLE
PDOException: There is no active transaction in ...

### ISSUE THREAD

**Issue by AndyAlvarezDev** [centralnicgroup-opensource/rtldev-middleware-whmcs#233]
In the "TLD Import & Pricing Sync" section of whmcs I cannot update the price list of Hexonet ispapi, it returns the error "There is no active transaction"...

WHMCS 8.6.0

[Screenshost 1](https://prnt.sc/hAs44oUWcw5w)

**Comment by KaiSchwarz-cnic** [comment]
Hi @AndyAlvarezDev 

We upgraded as well to WHMCS 8.6.0, but it works fine on our end.

a) which DB Software + Version do you have in use
b) when does this error exactly happen? when loading the List of TLDs or when Importing / Updating the TLDs and their prices?

Best
Kai

**Comment by KaiSchwarz-cnic** [comment]
@AndyAlvarezDev please be so kind as to provide the above information. Otherwise we cannot help a lot with your support ticket.
In addition I can forward that we noticed WHMCS 8.6.0 itself not being that stable at least with PHP8.
This could be another such case, not sure. Still, it works fine for us.

Our Dev-Setup is (just for reference)
* Ubuntu 22.04.1 LTS
* MySQL 8
* Apache 2.4.52 (LiteSpeed alternatively)
* PHP 7.4 / PHP 8

HTH
Kai

**Comment by KaiSchwarz-cnic** [comment]
As of missing feedback, I've to close this issue. Still, feel free to reply and provide the requested information -  we'll reopen the ticket then.

**Comment by KaiSchwarz-cnic** [comment]
Addressed this reproducible issue to WHMCS Support. Looks like with PHP 8.1 they upgraded a DB Library and parts of our db communication is broken therefore.

It is still documented that way here under [Getting to PDO](https://developers.whmcs.com/advanced/db-interaction/).
In parallel we are working on the analysis on our end to ensure this is patched as fast as possible.

**Comment by KaiSchwarz-cnic** [comment]
WHMCS Support provided a snippet to use and we rolled out version [16.6.3](https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/releases/tag/v16.6.3).

Just in case this isn't patching that issue, please let us know.

**Comment by KaiSchwarz-cnic** [comment]
another issue I noticed with PHP8 and the pricing sync got patched in [16.6.4](https://github.com/centralnicgroup-opensource/rtldev-middleware-whmcs/releases/tag/v16.6.4).

The Registrar TLD Sync is now working as expected. Sorry for the inconveniences caused!

Kai

**Comment by KaiSchwarz-cnic** [comment]
16.6.5 is patching all the remaining PHP 8 issues I noticed on my end.
